### ISSUE TITLE
oslc un-escapes the path to stdosl.h

### ISSUE THREAD

**Issue by LazyDodo** [imageworks/OpenShadingLanguage#866]
### Problem

When installing oslc in a path that has \n in it ( such as c:\osl_test\new\ ) oslc will
fail trying to compile any shader. due to the path being un-escaped somewhere. 

### Steps to Reproduce

```
c:\osl_test>new\osl\bin\oslc.exe test.osl
c:\osl_test\test.osl:1: error: error: could not find include file: c:\osl_test
ew\osl/shaders/stdosl.h

FAILED test.osl

c:\osl_test>rename new old

c:\osl_test>old\osl\bin\oslc.exe test.osl
Compiled test.osl -> mytest.oso

c:\osl_test>
```
I validated with procmon that the path is indeed unescaped

### Versions

* OSL branch/version: tested 1.7.5 and 1.9.3
* OS: Windows
* C++ compiler: msvc 2015

**Comment by lgritz** [comment]
Hi, sorry for this to slip through the cracks. Can somebody verify whether or not this is still a problem in the current master and/or release branch?

**Comment by lgritz** [comment]
#1073 seems to be on a related topic, but I'm not sure if it addresses this particular bug.

**Comment by LazyDodo** [comment]
Not a problem! Between the time of reporting and now I'm only aware of a single blender user bumping into this one. 

I can still reproduce the issue 
```
c:\osl>old\bin\oslc.exe test.osl
Compiled test.osl -> test.oso

c:\osl>rename old new

c:\osl>new\bin\oslc.exe test.osl
c:\osl\test.osl:1: error: error: could not find include file: c:\osl
ew/shaders/stdosl.h
FAILED test.osl

c:\osl>new\bin\oslc.exe --help
oslc -- Open Shading Language compiler 1.10.8
(c) Copyright 2009-2019 Sony Pictures Imageworks, et al. All rights reserved.
```

**Comment by lgritz** [comment]
So the problem is specifically when the path to oslc contains a directory starting with n?

Or is it also a problem when the osl file is in a directory starting with n?

um, I mean,

    > c:\path\to\new\oslc test.osl

    > oslc c:\path\to\new\test.osl

Both problems, just first, or just second?

**Comment by lgritz** [comment]
Proposed fix in #1101.

Would you by any chance be able to test this on Windows?
If it works, I can backport to release branch as well.

**Comment by LazyDodo** [comment]
That worked, with a minor caveat, i build against an rather old version oiio (1.8.13) which seems to be missing `OIIO::Strutil::sprintf` replaced it with `OIIO::Strutil::format` and that did the trick.

**Comment by lgritz** [comment]
Closing, should be fixed by #1101. It's in master now, and I will backport to 1.10, pending tests.
